x/staking/client/cli/tx.go

```diff
@@ -1,26 +1,406 @@
 package cli
 
 import (
+	"bytes"
+	"encoding/hex"
+	"errors"
 	"fmt"
 
 	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
 
 	"github.com/cosmos/cosmos-sdk/client"
-	// "github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/maticnetwork/bor/common"
+
+	// "github.com/maticnetwork/heimdall/bridge/setu/util"
+	"github.com/maticnetwork/heimdall/contracts/stakinginfo"
+	"github.com/maticnetwork/heimdall/helper"
+	hmTypes "github.com/maticnetwork/heimdall/types/common"
 	"github.com/maticnetwork/heimdall/x/staking/types"
 )
 
+var logger = helper.Logger.With("module", "staking/client/cli")
+
 // GetTxCmd returns the transaction commands for this module
 func GetTxCmd() *cobra.Command {
-	cmd := &cobra.Command{
+	stakingTxCmd := &cobra.Command{
 		Use:                        types.ModuleName,
 		Short:                      fmt.Sprintf("%s transactions subcommands", types.ModuleName),
 		DisableFlagParsing:         true,
 		SuggestionsMinimumDistance: 2,
 		RunE:                       client.ValidateCmd,
 	}
 
-	// this line is used by starport scaffolding # 1
+	stakingTxCmd.AddCommand(
+		SendValidatorJoinTxCmd(),
+		SendStakeUpdateTxCmd(),
+		SendSignerUpdateTxCmd(),
+		SendValidatorExitTxCmd(),
+	)
+
+	return stakingTxCmd
+}
+
+// SendValidatorJoinTx send validator join transaction
+func SendValidatorJoinTxCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "validator-join",
+		Short: "Join Heimdall as a validator",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			clientCtx, err := client.ReadTxCommandFlags(clientCtx, cmd.Flags())
+			if err != nil {
+				return err
+			}
+
+			// get proposer
+			proposer := hmTypes.HexToAccAddress(viper.GetString(FlagProposerAddress))
+			if proposer.Empty() {
+				proposer = helper.GetFromAddress(clientCtx)
+			}
+
+			txhash := viper.GetString(FlagTxHash)
+			if txhash == "" {
+				return fmt.Errorf("transaction hash is required")
+			}
+
+			pubkeyStr := viper.GetString(FlagSignerPubkey)
+			if pubkeyStr == "" {
+				return fmt.Errorf("pubkey is required")
+			}
+
+			pubkeyBytes := common.FromHex(pubkeyStr)
+			if len(pubkeyBytes) != 65 {
+				return fmt.Errorf("Invalid public key length")
+			}
+			pubkey := hmTypes.NewPubKey(pubkeyBytes)
+
+			// total stake amount
+			amount, ok := sdk.NewIntFromString(viper.GetString(FlagAmount))
+			if !ok {
+				return errors.New("Invalid stake amount")
+			}
+
+			contractCallerObj, err := helper.NewContractCaller()
+			if err != nil {
+				return err
+			}
+			//ToDO
+			// chainmanagerParams, err := util.GetChainmanagerParams(cliCtx)
+			// if err != nil {
+			// 	return err
+			// }
+
+			// get main tx receipt  NOTE:// Change MaxConformation, Now it is hard coded.
+			receipt, err := contractCallerObj.GetConfirmedTxReceipt(hmTypes.HexToHeimdallHash(txhash).EthHash(), 6)
```

Comment:
add an issue to use config for `MinConfirmationCount` or something better

Reply:
We have `MinConfirmationCount` in `heimdall-config.toml`. If not we should create an issue.

Reply:
Cool, I have created an issue for that.
 https://github.com/maticnetwork/heimdall/issues/459